apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -0,0 +1,514 @@
+            yield return new object[] { MaxText, "NaN", Math.Sign(-1) };
+            yield return new object[] { PgNumeric.MaxValue.ToString(), "NaN", Math.Sign(-1) };
+            yield return new object[] { (BigInteger.Pow(10, 147455) - 1).ToString(), "NaN", Math.Sign(-1) };
+            yield return new object[] { (-(BigInteger.Pow(10, 147455) - 1)).ToString(), "NaN", Math.Sign(-1) };
+            yield return new object[] { PgNumeric.MinValue.ToString(), "NaN", Math.Sign(-1) };
+            yield return new object[] { MinText, "NaN", Math.Sign(-1) };
+            yield return new object[] { "0", "NaN", Math.Sign(-1) };
+            yield return new object[] { "NaN", MaxText, Math.Sign(1) };
+            yield return new object[] { "NaN", MinText, Math.Sign(1) };
+            yield return new object[] { "NaN", "0", Math.Sign(1) };
+        }
+
+        private static string GetString(char character, int length)
+        {
+            return string.Concat(Enumerable.Repeat(character, length));
+        }
+
+        [Theory]
+        [InlineData("1.00", "1")]
+        [InlineData("1", "1.00")]
+        [InlineData("1", "1")]
+        [InlineData("NaN", "NaN")]
+        [InlineData("NAN", "Nan")]
+        [InlineData("0.00", "000")]
+        [InlineData("0.00", "0")]
+        [InlineData("1.0000", "1.000")]
+        [InlineData("1.00", "1.0")]
+        [InlineData("-1.00", "-1.0")]
+        [InlineData("0.00000009", "0.00000009")]
+        [InlineData("0.00000009", "0.00000009000000")]
+        [InlineData("0.00000009000000", "0.00000009")]
+        [InlineData("-0.00", "0.0")]
+        public void EqualsTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.Equals(number2);
+            Assert.Equal(true, result);
+        }
+
+        [Theory]
+        [InlineData("1.00", "1")]
+        [InlineData("1", "1.00")]
+        [InlineData("1", "1")]
+        [InlineData("NaN", "NaN")]
+        [InlineData("NAN", "Nan")]
+        [InlineData("0.00", "000")]
+        [InlineData("0.00", "0")]
+        [InlineData("1.0000", "1.000")]
+        [InlineData("1.00", "1.0")]
+        [InlineData("-1.00", "-1.0")]
+        [InlineData("0.00000009", "0.00000009")]
+        [InlineData("0.00000009", "0.00000009000000")]
+        [InlineData("0.00000009000000", "0.00000009")]
+        [InlineData("-0.00", "0.0")]
+        public void CompareToEqualValueTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.Equal(0, result);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetComparisionData))]
+        public void CompareToTest(string input1, string input2, int expectedResult)
```

Comment:
If this is just for comparing NaN, the the method should be called CompareWithNan and the data member should be called GetNaNComparisonData or similar.